### DISCUSSION TITLE
Feature request: non–stationarity

### DISCUSSION THREAD

**Discussion by kuchaale** [georgebv/pyextremes#11] [Ideas]
**Is your feature request related to a problem? Please describe.**
Would it be somehow possible to investigate non–stationarity, e.g. different seasons or long term trends (e.g. Chapter 6  in [Cole (2001)](https://www.springer.com/de/book/9781852334598))? 

**Describe the solution you'd like**
Please see e.g. R code [here](http://www.mas.ncl.ac.uk/~nlf8/teaching/mas8391/background/chapter5.pdf) where a a argument tells which column(s) of data use as covariates for the linear modelling of the location parameter.

Thanks for your answer

**Comment by georgebv** [comment]
@kuchaale thank you for creating the first thread on pyextremes!

How do you see interface looking for non-stationary processes? My understanding is that you apply some transformation to your data (e.g. de-trending) prior to using `EVA` and with non-stationary processes de-trending arguments (e.g. mean and slope for linear) would be part of the model. This way `extremes` series (the `.get_extremes` method) would be dynamic (changing for every iteration) so I cannot see it being a part of `EVA` - we'll need a new utility class altogether.

**Comment by jacktang** [comment]
@georgebv I think simple de-trending approach is problematic(?). The relationship between extremes and temporal context is lost, and extremes may be tied to specific temporal conditions. Also return level calculations become disconnected from the original sequences.

**Reply by georgebv** [reply]
Can you provide more context? AFAIK, the entire premise of GEVD/GPD is that you model distribution of maxima/exceedances of a random signal.

**Reply by jacktang** [reply]
@georgebv I tried to build extremes prediction model like answer the question: when do some scale extremes happen in future? Do you think GEV/GPD model can do the prediction?

**Reply by georgebv** [reply]
I don't understand the question - are you trying to predict specific time when an event will happen? EVA is about estimating probabilities of exceedance - probability of a given value being exceeded within a time period, I have a [documentation section about it](https://georgebv.github.io/pyextremes/user-guide/6-return-periods/).

**Reply by jacktang** [reply]
Thanks for the documentation link :) 
I am going to predict what extreme will happen in specific time. Say what scale of flood will happen in this summer. And by POT/GPD model, I try to make prediction on each hour (I suppose the records is hourly interval).

**Reply by georgebv** [reply]
Once you start predicting specific values at specific times you are not really dealing with statistics - that would be akin to predicting values of specific coin flips or dice rolls instead of their distribution. EVA gives you probability of making an observation exceeding a certain value at least once within a given interval. A numerical model implementing some Navier Stokes approximation would be a more appropriate tool for your problem.

**Reply by jacktang** [reply]
~~@georgebv you're right that the prediction and the EVA are two different things. And if i know the extreme values distribution, I can use some methods like mcmc to calculate the probability. And that's why I considered de-trending approach might be problematic.~~
